sources/api/prairiedog/src/main.rs

```diff
@@ -253,8 +284,126 @@ fn load_crash_kernel() -> Result<()> {
     Ok(())
 }
 
+/// Queries Bottlerocket boot settings and generates initrd image file with boot config as the only data
+pub(crate) async fn generate_boot_config<P>(socket_path: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let initrd = match get_boot_config_settings(socket_path).await? {
+        Some(boot_settings) => {
+            info!("Generating initrd boot config from boot settings");
+            trace!("Boot settings: {:?}", boot_settings);
+            let bootconfig = bootconfig::ser::to_string(&boot_settings)
```

Comment:
Consider re-exporting some functions from `ser` and `de` as a convenience, like the `toml` crate does:
```suggestion
            let bootconfig = bootconfig::to_string(&boot_settings)
```